### PR TITLE
Kl/stmdt counterpart fix EVM-701

### PR DESCRIPTION
**Pull request by kelemeno**
# What ❔

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.

### CODE REVIEW

File: l1-contracts/contracts/bridge/L1AssetRouter.sol [hunk 1]
```diff
@@ -255,41 +257,36 @@ contract L1AssetRouter is IL1AssetRouter, ReentrancyGuard, Ownable2StepUpgradeab
...
 
         bytes memory l2Calldata = abi.encodeCall(
             IL2Bridge.setAssetHandlerAddress,
-            (_assetId, _assetHandlerAddressOnCounterPart)
+            (_assetId, _assetHandlerAddressOnCounterpart)
         );
-
-        L2TransactionRequestDirect memory request = L2TransactionRequestDirect({
-            chainId: _chainId,
+        /// Note these L1->L2 txs cannot be claimed as failed, but we still need to give them a unique txDataHash
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/bridge/L1AssetRouter.sol:280]: why do we need unique hashes for those?

File: l1-contracts/contracts/bridgehub/STMDeploymentTracker.sol [hunk 2]
```diff
@@ -32,6 +32,13 @@ contract STMDeploymentTracker is ISTMDeploymentTracker, ReentrancyGuard, Ownable
         _;
     }
 
+    /// @notice Checks that the message sender is the bridgehub.
+    modifier onlyOwnerViaRouter(address _prevMsgSender) {
+        // solhint-disable-next-line gas-custom-errors
+        require(msg.sender == address(SHARED_BRIDGE) && _prevMsgSender == owner(), "STM DT: not owner via router");
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/bridgehub/STMDeploymentTracker.sol:38]: btw, maybe rename the `SHARED_BRIDGE` to `ASSET_ROUTER`? Or even `L1_ASSET_ROUTER`?

File: l1-contracts/contracts/bridge/L1AssetRouter.sol [hunk 3]
```diff
@@ -702,32 +706,32 @@ contract L1AssetRouter is IL1AssetRouter, ReentrancyGuard, Ownable2StepUpgradeab
         bytes memory _transferData,
         bytes32 _expectedTxDataHash
     ) internal view returns (bool isLegacyTxDataHash) {
-        try this.encodeTxDataHash(true, _prevMsgSender, _assetId, _transferData) returns (bytes32 txDataHash) {
+        try this.encodeTxDataHash(0x00, _prevMsgSender, _assetId, _transferData) returns (bytes32 txDataHash) {
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/bridge/L1AssetRouter.sol:709]: Could you please create named constants for encoding version `0x00`, `0x01` and `0x02`?

File: l1-contracts/test/foundry/unit/concrete/Bridges/L1SharedBridge/L1SharedBridgeFails.t.sol [hunk 4]
```diff
@@ -93,23 +93,23 @@ contract L1AssetRouterFailTest is L1AssetRouterTest {
...
-            eraChainId,
-            mintValue,
-            l2TxGasLimit,
-            l2TxGasPerPubdataByte,
-            refundRecipient,
-            tokenAssetId,
-            address(token)
-        );
-    }
+    // function test_setAssetHandlerAddressOnCounterpart_notOwnerOrADT() public {
```
Comment (StanislavBreadless) [inline l1-contracts/test/foundry/unit/concrete/Bridges/L1SharedBridge/L1SharedBridgeFails.t.sol:96]: commented out code

File: l1-contracts/contracts/bridge/L1AssetRouter.sol (hunk 1 above)
Comment (kelemeno) [reply l1-contracts/contracts/bridge/L1AssetRouter.sol:280]: True, we don't actually need it.

File: l1-contracts/test/foundry/unit/concrete/Bridges/L1SharedBridge/L1SharedBridgeFails.t.sol (hunk 4 above)
Comment (kelemeno) [reply l1-contracts/test/foundry/unit/concrete/Bridges/L1SharedBridge/L1SharedBridgeFails.t.sol:96]: Still need to fix this test

**Review by StanislavBreadless — APPROVED** [review]

**Pull request merged**